instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -123,38 +123,38 @@ def generated_postgres_regex
             @generated_postgres_regex ||= Regexp.union(PG::Constants::POSTGRES_COMPONENTS.map { |component| PG::Constants::COMPONENTS_REGEX_MAP[component] })
           end
 
-          def database_name
-            conninfo_hash[:dbname]&.to_s
-          end
-
-          def first_in_list(item)
-            if (idx = item.index(','))
-              item[0...idx]
-            else
-              item
-            end
-          end
-
           def client_attributes
             attributes = {
               'db.system' => 'postgresql',
-              'db.user' => conninfo_hash[:user]&.to_s,
-              'db.name' => database_name,
-              'net.peer.name' => first_in_list(conninfo_hash[:host]&.to_s)
+              'db.user' => user&.to_s,
+              'db.name' => db&.to_s,
+              'net.peer.name' => host&.to_s
             }
             attributes['peer.service'] = config[:peer_service] if config[:peer_service]
 
             attributes.merge(transport_attrs).reject { |_, v| v.nil? }
           end
 
+          def transport_addr
+            # The hostaddr method is available when the gem is built against
+            # a recent version of libpq.
+            return hostaddr if defined?(hostaddr)
+
+            # As a fallback, we can use the hostaddr of the parsed connection
+            # string when there is only one. Some older versions of libpq allow
+            # multiple without any way to discern which is presently connected.
+            addr = conninfo_hash[:hostaddr]&.to_s
+            return addr unless addr&.include?(',')
+          end
+
           def transport_attrs
-            if conninfo_hash[:host]&.start_with?('/')
+            if host&.start_with?('/')
               { 'net.transport' => 'Unix' }
             else
               {
                 'net.transport' => 'IP.TCP',
-                'net.peer.ip' => conninfo_hash[:hostaddr]&.to_s,
-                'net.peer.port' => first_in_list(conninfo_hash[:port]&.to_s)
+                'net.peer.ip' => transport_addr&.to_s,
+                'net.peer.port' => port&.to_s
```

Comment:
🤔 I'm curious about this line. Semantic Conventions say that `peer.port` should be an integer. Is `to_s` appropriate here?

- https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/span-general/

Reply:
I don't actually recall what the reasons were for stringification here; which suggests that perhaps there was no valid reason and it was just a mistake. If you want to include a fix for this in the PR, that would be a welcome change. 😄